scrabble.js

```diff
@@ -1,16 +1,165 @@
+
+    if (wordLetters.length > 7) {
+      throw "Character is not a letter!";
+    } else if (wordLetters.length === 0) {
+      throw "Oops! This word is empty!";
+    }
 
+    wordLetters.forEach(function(letter) {
+      if (LETTERS.test(letter)) {
+        score += WORDVALUES[letter];
+      } else {
+        throw "This contains a character is not a letter!";
+      }
+    });
+
+    if (wordLetters.length === 7) {
+      score += 50;
+    }
+    return score;
+  },
+
+  splitString(upWord) {
+    let wordLetters = upWord.split('');
+    return wordLetters;
   },
-  highestScoreFrom(arrayOfWords) {
 
+  highestScoreFrom(arrayOfWords) {
+    if (Array.isArray(arrayOfWords)) {
+      if (arrayOfWords.length === 0) {
+        throw "Oops! No words were given!";
+      } else if (arrayOfWords.length === 1) {
+        return arrayOfWords[0];
+      } else {
+        let topWord = arrayOfWords[0];
+        for (let i = 1; i < arrayOfWords.length; i += 1) {
+          if (this.score(arrayOfWords[i]) > this.score(topWord)) {
+            topWord = arrayOfWords[i];
+          } else if (this.score(arrayOfWords[i]) === this.score(topWord)) {
+            if (arrayOfWords[i].length === 7  && topWord.length < 7 ) {
+              topWord = arrayOfWords[i];
+            }
+            else if ( topWord.length < 7 && topWord.length > arrayOfWords[i].length) {
+              topWord = arrayOfWords[i];
+            }
+          }
+          return topWord;
+        }
+      }
+    } else {
+      throw "No array of words given!";
+    }
   },
+
 };
 
 Scrabble.Player = class {
+  constructor (name) {
+    this.name = name;
+    this.plays = [];
+    if (name === undefined) {
+      throw "Oops! No name was given!";
+    }
+  }
 
-};
+  play(word) {
+    if (this.hasWon()) {
+      return false;
+    } else {
+      const wordUp = word.toUpperCase();
+      let wordLetters = wordUp.split('');
+      if (wordLetters.length === 0) {
+        throw "No word was given!";
+      }
+      if (this.wordPrep(wordLetters)) {
+        this.plays.push(word);
+        return `You successfully played ${word}!`;
+      }
+    }
+  }
+
+  wordPrep (wordLetters) {
+    wordLetters.forEach(function(letter) {
+      if (!LETTERS.test(letter)) {
+        throw "This contains a character is not a letter!";
+      }
+    });
+    return true;
+  }
+
+  totalScore() {
+    let score = 0;
+    if (this.plays.length === 0) {
+      return score;
+    } else {
+      for (const word of this.plays) {
+        score += Scrabble.score(word);
+      }
+      return score;
+    }
+  }
+
+  hasWon() {
+    if (this.totalScore() < 100) {
+      return false;
+    } else {
+      return true;
+    }
+  }
 
+  highestScoringWord() {
+    if (this.plays.length === 0) {
+      throw 'No words have been played!';
+    }
+    let bestWord = this.plays[0];
+    for (let i = 1; i <= this.plays.length -1; i += 1) {
+      if (Scrabble.score(this.plays[i]) > Scrabble.score(bestWord)) {
```

Comment:
You could optimize by storing the score for the `bestWord` variable as well - otherwise you'll potentially be recalculating the score of this variable many times